server/main.js

```diff
@@ -139,39 +142,23 @@ const uniqueUserGauge = new promClient.Gauge({
 
 app.use(async (req, res, next) => {
   try {
-    let auth = req.header('Authorization')
-    if (_.startsWith(auth, 'Bearer ')) {
-      auth = auth.slice(7)
-    }
-
-    auth = _.split(auth, ':')
-    if (auth.length !== 2) {
-      writeError(res, 403, 'invalid Authorization token')
-      return
-    }
-
-    const keyID = auth[0]
-    const secretKey = auth[1]
-
-    const users = await Users.query('keyID').eq(keyID).exec()
-    const user = users[0]
-
-    if (!user) {
-      writeError(res, 403, 'invalid Authorization token')
-      return
-    }
-    if ((await hashPass(secretKey, user.salt)) !== user.hashedSecretKey) {
-      writeError(res, 403, 'invalid Authorization token')
-      return
-    }
-    let count = UNIQUE_USER_COUNT[keyID]
+    let count = UNIQUE_USER_COUNT[req.user.address]
     if (!count) {
       count = 0
     }
-    UNIQUE_USER_COUNT[keyID] = count + 1
-    bundleCounterPerUser.inc({ username: keyID.slice(0, 8) })
+    UNIQUE_USER_COUNT[req.user.address] = count + 1
+
+    // todo: is first two words sufficient entropy?
+    const username = entropyToMnemonic(req.user.address).split(' ').slice(0, 2).join('-')
+    bundleCounterPerUser.inc({ username })
```

Comment:
@jparyani this should generate a deterministic human readable name

I am wondering if it has sufficient entropy

I was looking at this as an alternative: https://github.com/Atrox/haikunatorjs

But the name generation is not deterministic